qemu/tests/virtio_fs_share_data.py

```diff
@@ -233,6 +236,11 @@ def get_viofs_exe(session):
                 session.cmd(cmd_pjdfstest % fs_dest, io_timeout)
 
             if cmd_xfstest:
+                # Increase tmpfs /dev/shm to 32G
```

Comment:
And since here is the method of directly converting to mem size without viewing the original dir size.
So it’s better to do it in the preprocess stage, like 107 line;
```
    if cmd_xfstest:
        ori_tmpfs_size = process.run(cmd_get_tmpfs, shell=True).stdout_text.replace("\n", "")
        logging.debug("original tmpfs size is %s", ori_tmpfs_size)
        if process.run(cmd_set_tmpfs % size_mem1, shell=True).exit_status:
            test.fail('xfstest set tmpfs failed.')
        if setup_local_nfs:
            for fs in params.objects("filesystems"):
                nfs_params = params.object_params(fs)
                params["export_dir"] = nfs_params.get("export_dir")
                params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
                params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
                nfs_local = nfs.Nfs(params)
                nfs_local.setup()
        params["start_vm"] = "yes"
        env_process.preprocess_vm(test, params, env, params["main_vm"])
```
@YongxueHong what do you think?

Reply:
> And since here is the method of directly converting to mem size without viewing the original dir size.
> So it’s better to do it in the preprocess stage, like 107 line;
> 
> ```
>     if cmd_xfstest:
>         ori_tmpfs_size = process.run(cmd_get_tmpfs, shell=True).stdout_text.replace("\n", "")
>         logging.debug("original tmpfs size is %s", ori_tmpfs_size)
>         if process.run(cmd_set_tmpfs % size_mem1, shell=True).exit_status:
>             test.fail('xfstest set tmpfs failed.')
>         if setup_local_nfs:
>             for fs in params.objects("filesystems"):
>                 nfs_params = params.object_params(fs)
>                 params["export_dir"] = nfs_params.get("export_dir")
>                 params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
>                 params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
>                 nfs_local = nfs.Nfs(params)
>                 nfs_local.setup()
>         params["start_vm"] = "yes"
>         env_process.preprocess_vm(test, params, env, params["main_vm"])
> ```
> 
> @YongxueHong what do you think?

If you don't check the current size of tmpfs but set tmpfs size directly, suggest you could put them into `pre_command` and `post_command` in the cfg to do that.

Reply:
Hi @zhencliu
Sorry, I wanna @zhenyzha

Reply:
Looks like this comment is for @zhenyzha . 

Agree, testing env setup in pre_command and cleanup in post_command are better(e.g. with some shell cmds), let's focus on the test assertions.

Reply:
Ok, got it. Thanks for your memtion.
Since some testing env setup codes already exists, i will move them to pre/post_command in a new commit.

Reply:
Anyway, your solution looks good to me. Up to you.